pcrmc/cli.py

```diff
@@ -79,6 +79,47 @@ def add_contact(name: List[str] = typer.Argument(...),
         )
 
 
+@app.command()
+def modify_contact(id: int = typer.Argument(...),
+                   field: str = typer.Option(str(), "--field", "-f"),
+                   value: str = typer.Option(str(), "--value", "-v")
+                   ) -> None:
+    """Modify contact by id."""
+    contacter = get_contacter()
+    _, error = contacter.modify_contact(id, field, value)
+
+    if error:
+        typer.secho(
+            f'modify_contact failed with "{ERRORS[error]}"',
+            fg=typer.colors.RED
+        )
+        raise typer.Exit(1)
+    else:
+        typer.secho(
+            f"pcrmc: Contact {id} modified",
```

Comment:
This prints the id that the user supplied, but maybe not the id which was actually modified. Assuming for some reason the backend messed up and modified the wrong id, the user would not be notified about this. Instead of printing the input, we should have some mechanism where the database in DPResponse includes the modified id, which then through ContacterResponse gets propagated here and is printed for the user.

Reply:
any ideas?

Reply:
One thought I had was to add more specific write functions in database.py, like write_all() (this is what we currently have) and write_contact() which appends or modifies a single given contact passed in as an object (maybe even make one function for append and one for modify, probably better). The Controller would still be responsible to assemble the object, then the database writes it (and deals with stuff like the ID). This way, the database can then return the object that it wrote (with ID and all) back to the controller and controller to cli.

Reply:
that would be ideal, had the same idea. implemented at least distinct read and write methods for contacts and meetings